plasmapy/particles/particle_class.py

```diff
@@ -2482,3 +2484,181 @@ def molecule(
 >>> def is_electron(particle: ParticleLike):
 ...     return particle == Particle("e-")
 """
+
+
+class Photon(AbstractParticle):
+    """
+    A class to represent a Photon.
+
+    Special case of an Abstract Particle with hardcoded mass,
+    charge, charge_number, spin and half_life to that of a
+    photon.
+
+    Parameters
+    ----------
+    quantity : ~astropy.units.Quantity, optional
+        The defining parameter of a Photon. It can be in the units of
+        Energy, Frequency, Momentum or Wavelength.
+
+    Raises
+    ------
+    ValueError
+        If the quantity is either not of allowed type or it's a negative quantity.
+
+    See Also
+    --------
+    ~plasmapy.particles.particle_class.Particle
+
+    Notes
+    -----
+    If the quantity is not specified, then the corresponding values of Energy,
+    Frequency, Momentum and Wavelength will be set to ``numpy.nan`` in
+    the appropriate units.
+
+    Examples
+    --------
+    >>> from astropy import units as u
+    >>> from plasmapy.particles import Photon
+    >>> photon = Photon(5 * u.m)
+    >>> photon.wavelength
+    <Quantity 5. m>
+    >>> photon.energy
+    <Quantity 3.97289171e-26 J>
+    >>> photon.momentum
+    <Quantity 1.32521403e-34 J s / m>
+    >>> photon.frequency
+    <Quantity 59958491.6 1 / s>
+    >>> photon.spin
+    1
+    """
+
+    def __init__(self, quantity: Quantity = None):
+        self._set_default_photon_properties()
+        if (quantity is not None) and (not self._validate_quantity(quantity)):
+            raise ValueError(
+                "Specify non negative quantity with units of Energy, wavelength, frequency or momentum."
+            )
+
+        if quantity is None:
+            self.energy = None
+            self.frequency = None
+            self.momentum = None
+            self.wavelength = None
+        else:
+            if self._isMomentum(quantity):
+                self.momentum = quantity.to(u.kg * u.m / u.s)
+                self.wavelength = const.h / self.momentum
+                self.frequency = const.c / self.wavelength
+                self.energy = const.h * self.frequency
+            else:
+                self.energy = quantity.to(u.J, equivalencies=u.spectral())
+                self.frequency = quantity.to(u.s**-1, equivalencies=u.spectral())
+                self.wavelength = quantity.to(u.m, equivalencies=u.spectral())
+                self.momentum = const.h / self.wavelength
+
+    def _set_default_photon_properties(self):
+        """Set default properties of a Photon."""
+        self._mass = 0 * u.kg
+        self._charge = 0 * u.C
+        self._charge_number = 0
+        self._spin = 1
+        self._half_life = np.inf * u.s
```

Comment:
```suggestion
        self._half_life = np.inf * u.s
        self._categories = {"boson", "stable", "uncharged"}
```
(related to another suggestion below)